### PR TITLE
[#7050] Fix statement leak in checkPermissionByObjType (4-2-stable)

### PR DESCRIPTION
**Pull request by dworkin**
A fix for the statement leak in `checkPermissionByObjType`. This also solves the memory leak which I erroneously reported as a statement leak, before.

### CODE REVIEW

**Review by alanking — COMMENTED** [review]
I think we are in agreement that this change is fine. I have also confirmed that it passes the core test suite. I am going to write a test as described by this comment to add to this before we merge it: https://github.com/irods/irods/issues/7050#issuecomment-1516615931

**Review by alanking — APPROVED** [review]
Added test and watched it fail (appropriately) without the changes in this PR, then added the changes in this PR, and it passed. Will await one more approval

File: scripts/irods/test/test_all_rules.py [hunk 1]
```diff
@@ -1587,6 +1587,58 @@ def test_msiDataObjChksum_with_admin_keyword__issue_6118(self):
...
+            # error. Otherwise, this will have no output.
+            groupie_user.assert_icommand(['irule', '-r', rep_instance, rule_text, 'null', 'ruleExecOut'])
+
+        finally:
+            self.admin.assert_icommand(['ils', '-AL', os.path.dirname(logical_path)], 'STDOUT') # debugging
+
+            for i in range(group_count):
+                group_name = 'group_' + str(i + 1)
+                self.admin.assert_icommand(['iadmin', 'rfg', group_name, groupie_user.username])
+                self.admin.assert_icommand(['iadmin', 'rmgroup', group_name])
```
Comment (korydraughn) [inline scripts/irods/test/test_all_rules.py:1639]: Should these calls use `run_icommand` instead?

Comment (alanking) [reply scripts/irods/test/test_all_rules.py:1639]: For this case, I don't think it necessarily matters, but I will change them.

Comment (alanking) [reply scripts/irods/test/test_all_rules.py]: Done

**Review by korydraughn — APPROVED** [review]
Squash, pound, and merge when ready.

**Pull request merged**